src/applications/apps.ts

```diff
@@ -111,11 +111,20 @@ export function getAppStatus(appName): AppOrParcelStatus | null {
 let startWarningInitialized: boolean = false;
 
 export function registerApplication<ExtraProps extends CustomProps = {}>(
-  appNameOrConfig: string | RegisterApplicationConfig,
-  appOrLoadApp: Application,
+  RegisterApplicationConfig,
+): void;
+export function registerApplication<ExtraProps extends CustomProps = {}>(
+  appName: string,
+  app: Application,
   activeWhen: Activity,
+  customProps: ExtraProps | CustomPropsFn<ExtraProps>,
+): void;
+export function registerApplication<ExtraProps extends CustomProps = {}>(
+  appNameOrConfig: string | RegisterApplicationConfig,
+  appOrLoadApp?: Application,
```

Comment:
This is now an optional argument, which fixes a types issue I discovered while writing single-spa-vue tests

<img width="944" alt="image" src="https://github.com/user-attachments/assets/24112baa-1601-40ef-a011-a4aa35c77d0f" />